### PR TITLE
grt: add vias when computing obstructions from nets

### PR DESCRIPTION
**Pull request by eder-matheus**

### CODE REVIEW

File: src/grt/src/GlobalRouter.cpp [hunk 1]
```diff
@@ -3430,35 +3423,49 @@ void GlobalRouter::findNetsObstructions(odb::Rect& die_area)
...
-                  logger_->warn(GRT,
-                                41,
-                                "Net {} has wires outside die area.",
-                                db_net->getConstName());
-              }
-              odb::dbTechLayer* tech_layer = pshape.shape.getTechLayer();
-              applyObstructionAdjustment(obstruction_rect, tech_layer);
+          const odb::dbShape& shape = pshape.shape;
+          if (shape.isVia()) {
+            odb::dbTechLayer* tech_layer = shape.getTechLayer();
```
Comment (maliberty) [inline src/grt/src/GlobalRouter.cpp:3428]: This isn't right as it is for a segment not a via.  I think you want getViaBoxes.

**Review by maliberty — CHANGES_REQUESTED** [review]

**Review by maliberty — APPROVED** [review]
The code looks fine but I suggest a secure CI

**Comment by maliberty** [comment]
I started a secure CI for you (secure-grt_vias)

**Comment by eder-matheus** [comment]
PR https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/pull/1924 created to fix CI failures.

**Pull request merged**